UnitsNet.Tests/CustomCode/AreaTests.cs

```diff
@@ -90,5 +90,19 @@ public void Constructor_UnitSystemSI_AssignsSIUnit()
             var area = new Area(1.0, UnitSystem.SI);
             Assert.Equal(AreaUnit.SquareMeter, area.Unit);
         }
+
+        [Fact]
+        public void As_ImperialGivenSIUnitSystem_ReturnsSIValue()
+        {
+            var imperialArea = new Area(2.0, AreaUnit.SquareInch);
+            Assert.Equal(0.00129032, imperialArea.As(UnitSystem.SI));
+        }
+
+        [Fact]
+        public void ToUnit_ImperialGivenSIUnitSystem_ReturnsSIValue()
```

Comment:
It will pass only for a unit system that has a base unit in meters. You could define one in feet and it would have a different quantity value. That conversion is handled by other tests, but I added it here regardless. I can change it for sure.

Reply:
Wait, I forgot this one before merging.

Are you saying that this test will only pass because `ToUnit(UnitSystem.SI)` converts to the Meter unit?
I think that is false, because my the Area equality converts to the default/base unit before comparing values.

As an example: `Length.FromCentimeters(100).Equals(Length.FromMeters(1))` is `true`.

This is why I think this test does not actually test the  `ToUnit` behavior at all - because it could have incorrectly converted to nautical mile unit and it would still equal the left hand side.